src/lib/services/addon-service.ts

```diff
@@ -105,6 +105,18 @@ export default class AddonService {
             this.fetchAddonConfigs().then((addonInstances) => {
                 addonInstances
                     .filter((addon) => addon.events.includes(eventName))
+                    .filter(
+                        (addon) =>
+                            !addon.projects ||
+                            addon.projects.length == 0 ||
+                            addon.projects.includes(event.project),
+                    )
+                    .filter(
+                        (addon) =>
+                            !addon.environments ||
+                            addon.environments?.length == 0 ||
```

Comment:
Due to the first check on the condition I don't think we need optional chaining here. If we do, we should also add it to `addon.projects.length == 0 ||`
```suggestion
                            addon.environments.length == 0 ||
```